src/modules/entity/reducer.ts

```diff
@@ -0,0 +1,59 @@
+import { DeploymentWithMetadataContentAndPointers } from 'dcl-catalyst-client'
+import { loadingReducer, LoadingState } from 'decentraland-dapps/dist/modules/loading/reducer'
+import {
+  FetchEntitiesFailureAction,
+  FetchEntitiesRequestAction,
+  FetchEntitiesSuccessAction,
+  FETCH_ENTITIES_FAILURE,
+  FETCH_ENTITIES_REQUEST,
+  FETCH_ENTITIES_SUCCESS
+} from './actions'
+
+export type EntityState = {
+  data: Record<string, DeploymentWithMetadataContentAndPointers>
+  loading: LoadingState
+  error: string | null
+}
+
+const INITIAL_STATE: EntityState = {
+  data: {},
+  loading: [],
+  error: null
+}
+
+type EntityReducerAction = FetchEntitiesRequestAction | FetchEntitiesSuccessAction | FetchEntitiesFailureAction
+
+export function entityReducer(state: EntityState = INITIAL_STATE, action: EntityReducerAction): EntityState {
+  switch (action.type) {
+    case FETCH_ENTITIES_REQUEST: {
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        error: null
+      }
+    }
+    case FETCH_ENTITIES_SUCCESS: {
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        error: null,
+        data: {
+          ...state.data,
+          ...action.payload.entities.reduce((obj, entity) => {
```

Comment:
You can always one line this with ```
=> ({...obj, [entity.entityId]: entity })
```

Reply:
And even go the extra mile by doing 
```
data: action.payload.entities.reduce((obj, entity) => ({...obj, [entity.entityId]: entity }), state.data)
```

Reply:
Regarding the usage of `state.data` as the initial value of the reduce, that only works when using the spread, but if I'd do that without using the spread within the reduce, it would be mutating

Reply:
Oh wait haha it's not 23 seconds (i spread the wrong object 😅  ) it's 71 ms, but still way slower

Reply:
10k entries might sound like a lot, but we have reducer with 10x that (like the tiles on the map), I learned this the hard way once we added a spread in on reducer or selector that handled that amount of entries and it did freeze the UI for a couple seconds every time it got computed lol